### PR TITLE
Wishlist

### PR DESCRIPTION
**Pull request by Sameony**
Added wishlist functionality 

https://user-images.githubusercontent.com/80711125/161816017-1a47c95d-33de-4e96-8f05-09a13f070fe3.mp4

### CODE REVIEW

File: src/components/addRemoveButtons.jsx [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
-const AddRemoveButtons = () => {
+const AddRemoveButtons = (defaultState) => {
+  const {dispatchWish} =useWishlist();
   return (
     <>
         <button className="btn btn-success mg-y-sm">
             <i className="fas fa-shopping-cart mg-r-sm"></i>Add to Cart
           </button>
-          <button className="btn btn-caution mg-y-sm">
+          <button className="btn btn-caution mg-y-sm" onClick={()=>dispatchWish({type:"WISH_REMOVE",payload:defaultState.defaultState})}>
```
Comment (shuntzhere) [inline src/components/addRemoveButtons.jsx:11]: on dispatch you can just send the defaultState directly.

File: src/components/header.jsx [hunk 2]
```diff
@@ -23,13 +25,13 @@ const NavBar = () => {
...
           </Link>
-          <div className="fs-sm badge badgeTopRight badgeRed">2</div>
+          <div className="fs-sm badge badgeTopRight badgeRed">8</div>
         </div>
         <div className="badgeWrap mg-md">
           <Link to="/wishlist">
             <div className="fs-xl fas fa-heart"></div>
           </Link>
-          <div className="fs-sm badge badgeTopRight badgeRed">9+</div>
+          <div className="fs-sm badge badgeTopRight badgeRed">{state.wishlistItems.length>9?("9+"):state.wishlistItems.length}</div>
```
Comment (shuntzhere) [inline src/components/header.jsx:34]: You may remove the whole 9+ parts, as in the beginning there would be no product in the cart.

File: src/components/productCard.jsx [hunk 3]
```diff
@@ -1,13 +1,28 @@
...
     const rounded = Math.round(cardRating*2)/2
     const hasDecimal = rounded%1!==0;
     const roundedWhole = Math.floor(rounded);
     const stars = hasDecimal?roundedWhole+1:roundedWhole;
+    
+    const defaultState = {
+        id : id,
+        imageUrl: img,
+        title: cardTitle,
+        price: Price,
```
Comment (shuntzhere) [inline src/components/productCard.jsx:16]: keep this in lowercase.

**Review by shuntzhere — APPROVED** [review]
Code looks great, I like the minimal things you have implemented as extras. Great work.

File: src/components/productCard.jsx (hunk 3 above)
Comment (Sameony) [reply src/components/productCard.jsx:16]: i have updated it in the next PR, thanks

File: src/components/header.jsx (hunk 2 above)
Comment (Sameony) [reply src/components/header.jsx:34]: yes i handled that in the cartHandler branch

File: src/components/addRemoveButtons.jsx (hunk 1 above)
Comment (Sameony) [reply src/components/addRemoveButtons.jsx:11]: Thanks for pointing out, i will refactor it after its completed

**Review by nakulsharma15 — APPROVED** [review]
Looks good to me. No changes suggested. I Liked the folder structure and the way you named the variables and functions

**Pull request merged**